packages/types/lib/change-case/camel-to-kebab-case-object-key.ts

```diff
@@ -0,0 +1,12 @@
+import CamelToKebabCase from "./camel-to-kebab-case";
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type CamelToKebabCaseObjectKey<T> = T extends object
+  ? {
+      [K in keyof T as CamelToKebabCase<K & string>]: CamelToKebabCaseObjectKey<
+        T[K]
+      >;
+    }
+  : T;
```

Comment:
```suggestion
type CamelToKebabCaseObjectKey<T extends object> = {
      [K in keyof T as CamelToKebabCase<K & string>]: CamelToKebabCaseObjectKey<T[K]>;
    };
```
처럼 generic 타입 자체에 object 제한을 거는게 조금 더 안전하지 않을까요??
나머지 타입도 동일합니다

Reply:
추가로 object타입이 밴타입이니 다른 타입을 이용하는게 좋아보여요 `JsonObject`타입같은건 어떠신가요?

Reply:
JsonObject로 하는것이 좋겠네요. 재한을 걸지 않는 이유는
재귀적으로 호출이 가능하게 하기 위해서 입니다.

예를 들어 
{
  wantSnakeCase: Date
}

와 같이 있을 경우에도 변환이 가능한것이 좋을거 같아서요